src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -184,187 +184,248 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
-        organization.Use2fa = newPlan.Has2fa;
-        organization.UseApi = newPlan.HasApi;
-        organization.UseSso = newPlan.HasSso;
-        organization.UseKeyConnector = newPlan.HasKeyConnector;
-        organization.UseScim = newPlan.HasScim;
-        organization.UseResetPassword = newPlan.HasResetPassword;
-        organization.SelfHost = newPlan.HasSelfHost;
-        organization.UsersGetPremium = newPlan.UsersGetPremium || upgrade.PremiumAccessAddon;
-        organization.UseCustomPermissions = newPlan.HasCustomPermissions;
-        organization.Plan = newPlan.Name;
+        organization.PlanType = passwordManagerPlan.Type;
+        organization.Seats = (short)(passwordManagerPlan.BaseSeats + upgrade.AdditionalSeats);
+        organization.MaxCollections = passwordManagerPlan.MaxCollections;
+        organization.UseGroups = passwordManagerPlan.HasGroups;
+        organization.UseDirectory = passwordManagerPlan.HasDirectory;
+        organization.UseEvents = passwordManagerPlan.HasEvents;
+        organization.UseTotp = passwordManagerPlan.HasTotp;
+        organization.Use2fa = passwordManagerPlan.Has2fa;
+        organization.UseApi = passwordManagerPlan.HasApi;
+        organization.SelfHost = passwordManagerPlan.HasSelfHost;
+        organization.UsePolicies = passwordManagerPlan.HasPolicies;
+        organization.MaxStorageGb = !passwordManagerPlan.BaseStorageGb.HasValue ?
+            (short?)null : (short)(passwordManagerPlan.BaseStorageGb.Value + upgrade.AdditionalStorageGb);
+        organization.UseGroups = passwordManagerPlan.HasGroups;
+        organization.UseDirectory = passwordManagerPlan.HasDirectory;
+        organization.UseEvents = passwordManagerPlan.HasEvents;
+        organization.UseTotp = passwordManagerPlan.HasTotp;
+        organization.Use2fa = passwordManagerPlan.Has2fa;
+        organization.UseApi = passwordManagerPlan.HasApi;
+        organization.UseSso = passwordManagerPlan.HasSso;
+        organization.UseKeyConnector = passwordManagerPlan.HasKeyConnector;
+        organization.UseScim = passwordManagerPlan.HasScim;
+        organization.UseResetPassword = passwordManagerPlan.HasResetPassword;
+        organization.SelfHost = passwordManagerPlan.HasSelfHost;
+        organization.UsersGetPremium = passwordManagerPlan.UsersGetPremium || upgrade.PremiumAccessAddon;
+        organization.UseCustomPermissions = passwordManagerPlan.HasCustomPermissions;
+        organization.Plan = passwordManagerPlan.Name;
         organization.Enabled = success;
         organization.PublicKey = upgrade.PublicKey;
         organization.PrivateKey = upgrade.PrivateKey;
+        organization.UsePasswordManager = true;
+
+        if (secretsManagerPlan != null && useSecretsManager)
+        {
+            organization.SmSeats = (short)(secretsManagerPlan.BaseSeats + upgrade.AdditionalSmSeats);
+            organization.SmServiceAccounts = (int)(secretsManagerPlan.BaseServiceAccount + upgrade.AdditionalServiceAccount);
```

Comment:
My instinct is that this should only store _additional_ service accounts, not total service accounts. Otherwise we have to do the maths to subtract out the base storage accounts each time we send this data to the UI, then add it back in when the value is received back from the client. It's also just doubling up on data.

I can see we store totals for PM seats + storage, but they don't work with "additional" values in the UI.

We could still do it either way; any opinion @cyprain-okeke @r-tome or @shane-melton ?

Reply:
We can store the additional service accounts only and add the base when is need from the plan to get the total